bot/cogs/verification.py

```diff
@@ -88,6 +88,13 @@
 Request = t.Callable[[discord.Member], t.Awaitable]
 
 
+class Limit(t.NamedTuple):
+    """Composition over config for throttling requests."""
+
+    batch_size: int  # Amount of requests after which to pause
+    sleep_secs: int  # Sleep this many seconds after each batch
```

Comment:
What's the justification for this data structure? It's only 2 arguments; they could easily be passed directly to the function.

Reply:
I'm not sure, I think I just felt like both ints configure one thing, so it made sense to me to group them together. It also keeps the function signature a little bit cleaner & shorter. Instead of documenting them in the `_send_request` docstring I can explain what the composition is, and then just refer to the type. They can be separate if you prefer.

I had originally created this:
```py
ThrottledRequest = t.Tuple[Request, Limit]
```
... but eventually abandoned the idea.

Reply:
The function does not currently support *not* receiving a `limit` but if it did then I think it'd really make sense to group them. Then it enforces that either both are provided or neither. Perhaps I was originally considering making it optional, but since both uses now are throttled it'd probably just be unnecessary.

Reply:
I don't feel strongly about it.